tests/testprogs/mountns_wrapper.c

```diff
@@ -0,0 +1,48 @@
+#define _GNU_SOURCE
+#include <sched.h>
+#include <sys/mount.h>
+#include <sys/stat.h>
+#include <libgen.h>
+#include <unistd.h>
+#include <string.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#define BUFFER_SIZE 255
+
+#define errExit(msg)    do { perror(msg); exit(EXIT_FAILURE); \
+                        } while (0)
+
+
+
+int main(int argc, char *argv[])
+{
+
+    char* private_mount = "/tmp/unshare/";
+
+    if (unshare(CLONE_NEWNS | CLONE_FS) != 0)
+      errExit("Failed to unshare");
+
+    if (mount(NULL, "/", NULL, MS_REC | MS_PRIVATE, NULL) != 0)
+      errExit("Failed to make mount private");
+
+    if (mkdir(private_mount, S_IRWXG) != 0 && (errno != EEXIST))
+      errExit("Failed to make private mount dir");
+
+    char path[BUFFER_SIZE];
+    int size = readlink("/proc/self/exe", path, sizeof(path));
```

Comment:
`BUFFER_SIZE - 1` as your're writing to `path[size]` below.

Reply:
thanks